src/open_inwoner/openklant/services.py

```diff
@@ -952,37 +953,79 @@ def _kvk_list_param(kvk: str) -> PartijListParams:
             "soortPartij": "organisatie",
         }
 
-    @staticmethod
-    def _vestigingsnummer_list_param(vestigingsnummer: str) -> PartijListParams:
-        return {
-            "partijIdentificator__codeSoortObjectId": "vestigingsnummer",
-            "partijIdentificator__codeRegister": "hr",
-            "partijIdentificator__codeObjecttype": "vestiging",
-            "partijIdentificator__objectId": vestigingsnummer,
-            "soortPartij": "organisatie",
-        }
-
     def find_persoon_for_bsn(self, bsn: str) -> Partij | None:
         return self.find_partij_for_params(params=self._bsn_list_param(bsn))
 
-    def find_organisatie_for_kvk(self, kvk: str) -> Partij | None:
-        return self.find_partij_for_params(params=self._kvk_list_param(kvk))
-
-    def find_organisatie_for_vestigingsnummer(
-        self, vestigingsnummer: str
+    def find_organisatie_for_kvk_and_vestiging(
+        self,
+        *,
+        kvk: str,
+        vestigingsnummer: str | None = None,
     ) -> Partij | None:
-        return self.find_partij_for_params(
-            params=self._vestigingsnummer_list_param(vestigingsnummer)
+        parent_kvk_results = list(
+            self.client.partij_identificator.list_iter(
+                params={
+                    "partijIdentificatorCodeSoortObjectId": "kvk_nummer",
+                    "partijIdentificatorCodeRegister": "hr",
+                    "partijIdentificatorCodeObjecttype": "niet_natuurlijk_persoon",
+                    "partijIdentificatorObjectId": kvk,
+                }
+            )
         )
+        if not parent_kvk_results:
+            return None
 
-    def get_or_create_partij_for_user(
-        self, fetch_params: FetchParameters, user: User
-    ) -> tuple[Partij | None, bool]:
+        kvk_pi: PartijIdentificator
+        match len(parent_kvk_results):
+            case 0:
+                return None
+            case x if x > 1:
+                raise RuntimeError("Unexpectedly found multiple PIs for a single kvk")
```

Comment:
We normally do `logger.error` and then return `None` or similar. Is `RuntimeError` the right choice (especially since the error is not handled by the caller)?